### DISCUSSION TITLE
The 'set_topic_permissions' command doesn't work?

### DISCUSSION THREAD

**Discussion by sijiaQ** [rabbitmq/cluster-operator#717] [Q&A] [answered]
my user: user1
my exchange: exchange1
my routing key: routingKey1
I set the permissions :
rabbitmqctl set_topic_permissions -p / user1 exchange1 "routingKey2" "routingKey2"
but user1 can also produce and consume message.
Did I miss any Settings?

**Comment by Zerpet** [comment, marked as answer]
I replicated the setup you are describing, and I am getting an access refused error, as expected. Using the Java client, I observe this error:

```
channel error; protocol method: #method<channel.close>(reply-code=403, reply-text=ACCESS_REFUSED - access to topic 'routingKey1' in exchange 'exchange1' in vhost '/' refused for user 'user1', class-id=50, method-id=20)
```

Please also note that topic authorization, in the AMQP protocol, is implemented slightly different, as described here: https://www.rabbitmq.com/access-control.html#topic-authorisation

> In AMQP 0-9-1, for example, consumers consume from queues and thus the standard resource permissions apply. In addition for AMQP 0-9-1, binding routing keys between an AMQP 0-9-1 topic exchange and a queue/exchange are checked against the topic permissions configured, if any.

**Reply by sijiaQ** [reply]
Thank you so much ! it works well now.

**Comment by Zerpet** [comment]
I'm going to convert this issue to a discussion. You may get a notification that this issue has been closed, however, it will be moved to https://github.com/rabbitmq/cluster-operator/discussions
